addon/templates/components/materialize-collection-item.hbs

```diff
@@ -0,0 +1,13 @@
+{{#if isHeader}}
+  <h4>{{yield}}</h4>
+{{else}}
+  {{#if image}}
+    <img class="circle" alt="{{imageAlt}}" src="{{image}}"/>
+  {{/if}}
+  {{#if title}}
+    <span class="title">{{title}}</span>
+    <p>{{yield}}</p>
+  {{else}}
+    {{yield}}
+  {{/if}}
+{{/if}}
```

Comment:
Every conditional is a new view, and you have a lot of them here. A good first step is breaking out "header" collection items into a separate component, instead of making this single component switch between two modes based on `isHeader`